galley/pkg/config/analysis/analyzers/annotations/annotations.go

```diff
@@ -84,15 +83,26 @@ outer:
 		annotationDef := lookupAnnotation(ann)
 		if annotationDef == nil {
 			m := msg.NewUnknownAnnotation(r, ann)
-
-			if line, ok := util.ErrorLine(r, fmt.Sprintf(util.Annotation, ann)); ok {
-				m.Line = line
-			}
+			util.AddLineNumber(r, ann, m)
 
 			ctx.Report(collectionType, m)
 			continue
 		}
 
+		if annotationDef.Deprecated {
+			m := msg.NewDeprecatedAnnotation(r, ann)
+			util.AddLineNumber(r, ann, m)
+
+			ctx.Report(collectionType, m)
+		}
+
+		if annotationDef.FeatureStatus == annotation.Alpha {
```

Comment:
Is there a possibility that `FeatureStatus` can also be `Beta`? I can see a few suggestions in the API implementation https://github.com/istio/api/pull/1723#discussion_r516328816 https://github.com/istio/api/pull/1723#discussion_r516333241 https://github.com/istio/api/pull/1723#discussion_r519775401

Reply:
FeatureStatus can be beta.  However we have no way for users to supply "parameters" to analysis.  We can't say "report all statuses below level X".  In the future, when it is possible to operate Istio without using any Beta features, we can create a BetaAnnotation message and use it for that, because we can turn off individual messages.

I made this as a parameter because I believe we have another FeatureStatus level **before** alpha, `EXPERIMENTAL`, but the API doesn't reflect that.  @nmittler Do you think it makes sense to add `EXPERIMENTAL` to the enum?

Reply:
This is a separate effort ongoing to add experimental. Once that lands, we can add experimental check here.